xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/job/AbstractCopyOrMoveJob.java

```diff
@@ -53,8 +56,29 @@ public abstract class AbstractCopyOrMoveJob<T extends AbstractCopyOrMoveRequest>
     @Override
     protected void runInternal() throws Exception
     {
-        if (this.request.getDestination() != null) {
-            super.runInternal();
+        this.progressManager.pushLevelProgress(3, this);
+
+        try {
+            this.progressManager.startStep(this);
+            BeginFoldEvent beginEvent = getBeginEvent();
+            this.observationManager.notify(beginEvent, this, this.getRequest());
+            if (((CancelableEvent) beginEvent).isCanceled()) {
+                return;
+            }
+            this.progressManager.endStep(this);
+
+            this.progressManager.startStep(this);
+            if (this.request.getDestination() != null) {
+                super.runInternal();
+            }
+            this.progressManager.endStep(this);
+
+            this.progressManager.startStep(this);
+            EndFoldEvent endEvent = getEndEvent();
+            this.observationManager.notify(endEvent, this, this.getRequest());
+            this.progressManager.endStep(this);
```

Comment:
You don't need this endStep, it will be closed by popLevelProgress.

Reply:
I don't need it but it's not a mistake to have it right? Note that here it's some old code that I just moved, and personally I feel more comfortable to keep the symmetry start/end Step

Reply:
It's not going to cause problems, it's just useless.

> I feel more comfortable to keep the symmetry start/end Step

On my side, I usually prefer to limit plumbing to not pollute too much the "real" code.